### PR TITLE
Date creation not consistent with Magento/DateTime with french format

### PR DESCRIPTION
**Pull request by khodu**
### Description (*)

Resolved Date creation not consistent with Magento/DateTime with french format issue.
### Fixed Issues (if relevant)

1. magento/magento2#6187: Date creation not consistent with Magento/DateTime with french format

### Manual testing scenarios (*)

1.Try to format a date using \Magento\Framework\Stdlib\DateTime\DateTime->date("d/m/y", "15/08/2016")
2. Now Return correct value.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @khodu. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: lib/internal/Magento/Framework/Stdlib/DateTime/DateTime.php [hunk 1]
```diff
@@ -141,6 +141,9 @@ public function timestamp($input = null)
                 $result = $input->getTimestamp();
                 break;
             default:
+				if (strpos($input, '/') !== false) {
+                    $input = str_replace('/', '-', $input);
```
Comment (orlangur) [inline lib/internal/Magento/Framework/Stdlib/DateTime/DateTime.php:145]: From https://stackoverflow.com/a/2891949 it looks like day and month will be mixed up by such change. Thoughts?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Stdlib/DateTime/DateTime.php (hunk 1 above)
Comment (khodu) [reply lib/internal/Magento/Framework/Stdlib/DateTime/DateTime.php:145]: Now, Day and Month will not be mixed with new changes. please review and test it again.

Comment (khodu) [reply lib/internal/Magento/Framework/Stdlib/DateTime/DateTime.php:145]: Hello @orlangur,
I have done changes as per your suggestion. Can we resolve this conversation or you need any other update.

Comment (khodu) [reply lib/internal/Magento/Framework/Stdlib/DateTime/DateTime.php:145]: Hello @orlangur 
i have added below code
if (strpos($input, '/') !== false && substr($input,0,2) > 12) {
$input = str_replace('/', '-', $input);
can you please retest it and give me your feedback.

**Pull request closed without merging**